modules/light-clients/08-wasm/internal/ibcwasm/expected_interfaces.go

```diff
@@ -51,6 +51,24 @@ type WasmEngine interface {
 		deserCost wasmvmtypes.UFraction,
 	) ([]byte, uint64, error)
 
+	// Migrate will migrate an existing contract to a new code binary.
+	// This takes storage of the data from the original contract and the CodeID of the new contract that should
+	// replace it. This allows it to run a migration step if needed, or return an error if unable to migrate
+	// the given data.
+	//
+	// MigrateMsg has some data on how to perform the migration.
```

Comment:
Maybe I should improve the godocs here including mentioning explicitly that:

1. Migrate should run on the new code.
2. CodeHash change in client state is performed by the go code already (but wasm code can't do this currently as the code doesn't know its own code hash unless we pass it).

Open to more suggestions on what I should add

Reply:
I'd be perfectly happy leaving way wasmd has it